### ISSUE TITLE
[22.05] github-runner is outdated

### ISSUE THREAD

**Issue by yayayayaka** [NixOS/nixpkgs#200247]
### Describe the bug
github-runner 2.296.2 is outdated and isn't supported by github anymore.

### Steps To Reproduce
1. Enable `services.github-runner`
2. Register the runner with GitHub

### Expected behavior
The registered runner appears as "online" on GIthub

### Screenshots
```
$ journalctl -eu github-runner.service
[...]
Nov 08 19:44:47 build-worker-01 runsvc.sh[4610]: Starting Runner listener with startup type: service
Nov 08 19:44:47 build-worker-01 runsvc.sh[4610]: Started listener process, pid: 5111
Nov 08 19:44:48 build-worker-01 runsvc.sh[4610]: √ Connected to GitHub
Nov 08 19:44:48 build-worker-01 runsvc.sh[4610]: Current runner version: '2.296.2'
Nov 08 19:44:48 build-worker-01 runsvc.sh[4610]: 2022-11-08 19:44:48Z: Listening for Jobs
Nov 08 19:44:49 build-worker-01 runsvc.sh[4610]: An error occurred: Runner version v2.296.2 is deprecated and cannot receive messages.
Nov 08 19:44:49 build-worker-01 runsvc.sh[4610]: Runner listener exited with error code 2
Nov 08 19:44:49 build-worker-01 runsvc.sh[4610]: Runner listener exit with retryable error, re-launch runner in 5 seconds.
[...]
```

### Notify maintainers
@veehaitch @newam @kfollesdal @aanderse

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.0.7, NixOS, 22.05 (Quokka), 22.05pre-git`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - nixpkgs: `/etc/src/nixpkgs`
```

**Comment by newAM** [comment]
I had a PR open to fix this for 22.05, I assumed no one cared because it became outdated before anyone reviewed it: #196666

Not sure what the solution is.  Maybe we should remove the package from the stable channel and make it unstable only?

**Comment by aanderse** [comment]
I didn't get pinged on that PR because my name isn't in the maintainers list on the stable channel... only unstable. Please feel free to ping me for a merge every time. You can backport my name into the maintainers list if you want and I'll automatically get a notification. I really appreciate the work you do for this package :heart:

Thanks.

**Comment by newAM** [comment]
PR to fix is in #200283

**Comment by newAM** [comment]
#200283 is merged, will take a bit to appear in the channel: https://nixpk.gs/pr-tracker.html?pr=200283
